Extension/src/LanguageServer/configurations.ts

```diff
@@ -386,6 +391,70 @@ export class CppProperties {
         }
     }
 
+    public nodeAddonIncludesFound(): number {
+        return this.nodeAddonIncludes.length;
+    }
+
+    private async readNodeAddonIncludeLocations(rootPath: string): Promise<void> {
+        let error: Error | undefined;
+        let pdjFound: boolean = false;
+        const package_json: any = await fs.promises.readFile(path.join(rootPath, "package.json"), "utf8")
+            .then(pdj => {pdjFound = true; return JSON.parse(pdj); })
+            .catch(e => (error = e));
+
+        const pathToNode: string = which.sync("node");
+        const nodeAddonMap: { [dependency: string]: string } = {
+            "nan": `"${pathToNode}" --no-warnings -e "require('nan')"`,
+            "node-addon-api": `"${pathToNode}" --no-warnings -p "require('node-addon-api').include"`
+        };
+
+        if (!error) {
+            try {
+                for (const dep in nodeAddonMap) {
+                    if (dep in package_json.dependencies) {
+                        const execCmd: string = nodeAddonMap[dep];
+                        let stdout: string = await util.execChildProcess(execCmd, rootPath);
+                        if (!stdout) {
+                            continue;
+                        }
+
+                        // cleanup newlines
+                        if (stdout[stdout.length - 1] === "\n") {
+                            stdout = stdout.slice(0, -1);
+                        }
+                        // node-addon-api returns a quoted string, e.g., '"/home/user/dir/node_modules/node-addon-api"'.
+                        if (stdout[0] === "\"" && stdout[stdout.length - 1] === "\"") {
+                            stdout = stdout.slice(1, -1);
+                        }
+
+                        // at this time both node-addon-api and nan return their own directory so this test is not really
+                        // needed. but it does future proof the code.
+                        if (!await util.checkDirectoryExists(stdout)) {
+                            // nan returns a path relative to rootPath causing the previous check to fail because this code
+                            // is executing in vscode's working directory.
+                            stdout = path.join(rootPath, stdout);
+                            if (!await util.checkDirectoryExists(stdout)) {
+                                error = new Error(`${dep} directory ${stdout} doesn't exist`);
+                                stdout = '';
+                            }
+                        }
+                        if (stdout) {
+                            this.nodeAddonIncludes.push(stdout);
+                        }
+                    }
+                }
+            } catch (e) {
+                error = e;
+            }
+        }
+        if (error && pdjFound) {
```

Comment:
I think this should be something like
```
if (error) {
if (pdjFound) {
...
}
} else {
this.handleConfigurationChange();
}
```
i.e. to skip this.handleConfigurationChange when it's not needed.

Reply:
Or checking this.nodeAddonIncludes.length > 0 (i.e. in case one of the cases errors?).

Reply:
i think it is (conceivably) possible that both node-addon-api and nan are referenced in package.json, both requires succeed, but one of two util.checkDirectoryExists() fail.

but i am changing it as i think it's 1) not possible now given what node-addon-api and nan return and 2) a really tiny edge case going forward because it means the package has to return an invalid directory.

thank you.